agent/php_execute.c

```diff
@@ -1027,6 +991,20 @@ static void nr_php_execute_metadata_init(nr_php_execute_metadata_t* metadata,
   } else {
     metadata->function = NULL;
   }
+  if (!NRINI(code_level_metrics_enabled)
+      || ZEND_USER_FUNCTION != op_array->type) {
+    metadata->filepath = NULL;
+    return;
+  }
+  if (op_array->filename && op_array->filename->len) {
+    metadata->filepath = op_array->filename;
```

Comment:
Could we use `zend_copy_string()` to simplify code?

Reply:
Same comment: https://github.com/newrelic/newrelic-php-agent/pull/585#discussion_r1042762736